cms/djangoapps/contentstore/management/commands/swap_from_auto_track_cohort_pilot.py

```diff
@@ -0,0 +1,168 @@
+        if not audit_course_mode or not verified_course_mode:
+            raise CommandError('Audit or Verified course modes are not defined for course: %s' % course_id)
+
+        for item in items:
+            # Checks whether or not an xblock is published, taken from contentstore/views/item.py create_xblock_info
+            is_library_block = isinstance(item.location, LibraryUsageLocator)
+            published = modulestore().has_published_version(item) if not is_library_block else None
+
+            # Verify that there exists group access for this xblock, otherwise skip these checks
+            if item.group_access:
+                set_audit_enrollment_track = False
+                set_verified_enrollment_track = False
+
+                # Check the partition and group IDs for the audit course groups, if they exist in the xblock's access
+                # settings then set the audit track flag to true
+                for audit_course_user_group_partition_group in random_audit_course_user_group_partition_groups:
+                    audit_partition_group_access = item.group_access.get(
+                        audit_course_user_group_partition_group.partition_id,
+                        None
+                    )
+                    if (audit_partition_group_access
+                            and audit_course_user_group_partition_group.group_id in audit_partition_group_access):
+                        set_audit_enrollment_track = True
+
+                # Check the partition and group IDs for the verified course group, if it exists in the xblock's access
+                # settings then set the verified track flag to true
+                verified_partition_group_access = item.group_access.get(
+                    verified_course_user_group_partition_group.partition_id,
+                    None
+                )
+                if (verified_partition_group_access
+                        and verified_course_user_group_partition_group.group_id in verified_partition_group_access):
+                    set_verified_enrollment_track = True
+
+                # Add the enrollment track ids to a group access array
+                enrollment_track_group_access = []
+                if set_audit_enrollment_track:
+                    enrollment_track_group_access.append(audit_course_mode.id)
+                if set_verified_enrollment_track:
+                    enrollment_track_group_access.append(verified_course_mode.id)
+
+                # If either the audit track, or verified track needed to be updated set the access, update and publish
+                if set_verified_enrollment_track or set_audit_enrollment_track:
+                    # Check that the xblock is published if it needs changes, otherwise raise an error
+                    if not published:
+                        raise CommandError('XBlock for course, %s needs access changes, but is a draft' % course_id)
```

Comment:
We would need to raise much more specific information so the course team could figure out which xblocks are causing problems (and it would be great if we could get the information of all xblocks in this state with one run of the command to avoid iteratively running the command/fixing issues/running the command).